File: libtokens/src/escape_token.cc
#include "escape_token.h"

// Escape Tokens
const char APOSTROPHE = '\'';
const char BACKSLASH = '\\';
const char CARRIAGE_RETURN = '\r';
const char NEW_LINE = '\n';
const char NULL_CHARACTER = '\0';
const char QUOTATION_MARK = '\"';
const char TAB = '\t';

// Escape Token Array
const char escapes_array() {
  const char strings[7] = {
      APOSTROPHE,     BACKSLASH, CARRIAGE_RETURN, NEW_LINE, NULL_CHARACTER,
      QUOTATION_MARK, TAB,
  };

  return *strings;
}

File: fortran-tokens/src/escape_token.f90
module escape_token
  implicit none
    
  !> Escape Tokens
  character(len=1), parameter :: APOSTROPHE = "\'"
  character(len=1), parameter :: BACKSLASH = '\\'
  character(len=1), parameter :: CARRIAGE_RETURN = '\r'
  character(len=1), parameter :: NEW_LINE = '\n'
  character(len=1), parameter :: NULL_CHARACTER = '\0'
  character(len=1), parameter :: QUOTATION_MARK = '\"'
  character(len=1), parameter :: TAB = '\t'

end module escape_token

